python/pyspark/sql/utils.py

```diff
@@ -14,6 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+# To disallow implicit relative import. Remove this once we drop Python 2.
+from __future__ import absolute_import
```

Comment:
We need this. Otherwise, `import pandas` imports `pyspark/sql/pandas` if `pandas` is not installed. This is forbidden from Python 3.